### PR TITLE
Refactor FXIOS-5564 [v110] Refactor favicons in tabs

### PR DESCRIPTION
**Pull request by OrlaM**
[FXIOS-5564](https://mozilla-hub.atlassian.net/browse/FXIOS-5564)
#12896

### CODE REVIEW

File: Client/Frontend/Browser/Tab Management/Tab.swift [hunk 1]
```diff
@@ -786,21 +781,6 @@ class Tab: NSObject {
         }
         return .none
     }
-
-    func updateFaviconCache() {
```
Comment (nbhasin2) [inline Client/Frontend/Browser/Tab Management/Tab.swift:790]: Do we have a specific cache for the favicon? Or does it just use kingfisher's default cache

File: Storage/SQL/SQLiteQueue.swift [hunk 2]
```diff
@@ -18,7 +18,7 @@ open class SQLiteQueue: TabQueue {
     }
 
     fileprivate func factory(_ row: SDRow) -> ShareItem {
-        return ShareItem(url: row["url"] as! String, title: row["title"] as? String, favicon: nil)
+        return ShareItem(url: row["url"] as! String, title: row["title"] as? String)
```
Comment (nbhasin2) [inline Storage/SQL/SQLiteQueue.swift:21]: Q. As I recall we would store favicons in SQLite, did we get rid of that?

**Review by nbhasin2 — COMMENTED** [review]
lgtm, tried on sim as well

**Review by nbhasin2 — APPROVED** [review]

File: Client/Frontend/Browser/Tab Management/Tab.swift (hunk 1 above)
Comment (OrlaM) [reply Client/Frontend/Browser/Tab Management/Tab.swift:790]: Under the hood it uses kingfishers cache for the image and our own cache for the url. It's abstracted in a way that we can swap it out for our own implementation if we find the need to later on. See SiteImageCache for implementation details.

File: Storage/SQL/SQLiteQueue.swift (hunk 2 above)
Comment (OrlaM) [reply Storage/SQL/SQLiteQueue.swift:21]: Yes, we now store the favicon urls in just a basic data file, similar to how we store tabs today. We added an expiration mechanism so the file won't grow forever and should always be a manageable size. See FaviconURLCache for details.

**Pull request merged**